### PR TITLE
Add thumbnails to main feed

### PR DESCRIPTION
**Pull request by fmatosqg**

### CODE REVIEW

**Comment by fmatosqg** [comment]
currently it uses an easy to find url but it's too low res. It will take some extra work for looking through a better resolution image and putting it there.

File: lib/src/main.dart [hunk 1]
```diff
@@ -218,3 +219,26 @@ class MainPage extends StatelessWidget {
     );
   }
 }
+
+/// When thumbnail is available draw image using full width available.
+/// 
+/// TODO: replace thumbnail with hi-res image.
+Widget addImage(Submission submission) {
+  return submission.thumbnail.toString() != 'self'
```
Comment (mzegar) [inline lib/src/main.dart:227]: Can you move this check outside of this function? Just inline it on line 140.
Use `submission.isSelf` instead.

File: lib/src/main.dart [hunk 2]
```diff
@@ -218,3 +219,26 @@ class MainPage extends StatelessWidget {
     );
   }
 }
+
+/// When thumbnail is available draw image using full width available.
+/// 
+/// TODO: replace thumbnail with hi-res image.
+Widget addImage(Submission submission) {
```
Comment (mzegar) [inline lib/src/main.dart:226]: We shouldn't need a comment like `/// When thumbnail is available draw image using full width available.`, this should be explained in the code if possible.

something like `Widget addThumbnail` would be enough.

File: lib/src/main.dart [hunk 3]
```diff
@@ -218,3 +219,26 @@ class MainPage extends StatelessWidget {
...
+  return submission.thumbnail.toString() != 'self'
+      ? LayoutBuilder(
+          builder: (a, b) => CachedNetworkImage(
+                width: b.biggest.width,
+                imageUrl: submission.thumbnail.toString(),
+                fit: BoxFit.fitWidth,
+                placeholder: (context, url) => CircularProgressIndicator(),
+                errorWidget: (context, url, error) => Text(
+                  'Image not available',
```
Comment (mzegar) [inline lib/src/main.dart:235]: Can we just display nothing if it's an error widget? I feel like the `Image not available` isn't clean. Just return an empty `SizedBox()`

**Comment by mzegar** [comment]
> currently it uses an easy to find url but it's too low res. It will take some extra work for looking through a better resolution image and putting it there.

The addition is extremely good, I just can't stand the low res images. Could you figure out how to get high res images going? Alongside the feedback I provided.

File: lib/src/main.dart (hunk 2 above)
Comment (fmatosqg) [reply lib/src/main.dart:226]: Comments help find things faster and understand something that today is obvious after we forget the obvious parts. I'd rather add more comments to the existing code, and break existing functions so it's easier for me to find what I'm looking for.

File: lib/src/main.dart (hunk 3 above)
Comment (fmatosqg) [reply lib/src/main.dart:235]: Done.

**Comment by fmatosqg** [comment]
Planning on looking on higher res images next, but that may have to wait for next week.

**Pull request merged**